Clover/app/src/main/java/org/floens/chan/ui/controller/ImageViewerController.java

```diff
@@ -221,9 +221,15 @@ public void onSubMenuItemClicked(ToolbarMenuItem parent, FloatingMenuItem item)
                     public void onFloatingMenuItemClicked(FloatingMenu menu, FloatingMenuItem item) {
                         for (ImageSearch imageSearch : ImageSearch.engines) {
                             if (((Integer) item.getId()) == imageSearch.getId()) {
-                                final HttpUrl searchImageUrl = getSearchImageUrl(presenter.getCurrentPostImage());
-                                AndroidUtils.openLinkInBrowser((Activity) context, imageSearch.getUrl(searchImageUrl.toString()));
-                                break;
+                                if(imageSearch.getType()=="link") {
```

Comment:
You can't compare strings using `==`. But this will become an enum or const int anyway.